### PR TITLE
[CI/CD] Configure pipeline to deploy to GitHub Pages

### LINKED ISSUE #59: [CI/CD] Fix error and configure pipeline for GitHub pages deployment
### Objectives
There was an issue with the CI/CD pipeline that led to the `build-and-deploy` job to fail. Apparently, it cannot find the module of the jest-puppeteer. I investigated into it and it seems like the the job runs the exact same test as the `lint-and-test`, which we have already ran the tests. Additionally, the GitHub pages deployment hasn't been configured yet.

### Acceptance Criteria
- [ ] `build-and-deploy` job does not fail
- [ ] CI/CD pipeline builds and deploys to GitHub pages with a corresponding link

### Assignees
- @kennethkietvuong

### LINKED ISSUE #74: [Bug] Pipeline cannot find tests to run unit testing
### Objectives
I found a bug with the CI/CD pipeline with some local testing, and I found that the pipeline skips over the unit testing of the `lint-and-test` job because it cannot find any tests.

image

### Acceptance Criteria
- [x] CI/CD pipeline runs unit tests as expected
- [x] Test files found and ran properly to see test results

### Assignees
- Assignee 1
- Assignee 2
- ...

### Additional Notes <Optional>
Without looking further into it, I suspect that it might be an issue with either the pipeline itself or the way that the test script looks for the test files.

Potentially, one could investigate these files:

- `.github/ci.yml`
- `config/package.json`
- `__tests__/`

### PR DESCRIPTION
**Pull request by kennethkietvuong**
# Objective
The purpose of this PR is to fix and configure new changes to the existing CI/CD pipeline so that we are able to deploy GitHub pages for our project.

## Reason
1. Currently, our existing CI/CD pipeline has an issue with its `build-and-deploy` job that will essentially deploy the GitHub pages. I took a look at the error that occurs, and it seems like we are running unit tests when building/deploying. This doesn't make sense to me, since we have already ran through a `lint-and-test` job.

2. I also noticed that the deployment step for the CI/CD pipeline was not written out yet either, thus not deploying GitHub pages.

## Solution
My solution to these two reasons is to configure the pipeline file `ci.yml` to deploy a GitHub page with its corresponding link. Here is what I specifically did.

- Feature to manually run workflow from the `Actions` tab
- Modified the `build-and-deploy` job to deploy GitHub pages
  - Set permissions to allow deployment to GitHub pages
  - Concurrency setting (one deployment at a time)
 
## Post-Merge
If this PR gets merged, there is one thing that needs to be changed in the repository settings for the CI/CD pipeline to properly deploy GitHub pages. In the `Pages` tab, instead of `Deploy from a branch`, we should build and deploy from `GitHub Actions` instead.

## Extra Notes
With my changes, I wasn't too sure if we were using the `build` script.

### CODE REVIEW

**Comment by AliMohammadiasl** [comment]
Looks good to me!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,25 +47,27 @@ jobs:
       - name: Generate documentation
         run: npx jsdoc -c jsdoc.json
 
+  # Deploy GitHub pages
   build-and-deploy:
     needs: lint-and-test
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
```
Comment (adironene) [inline .github/workflows/ci.yml:55]: https://github.com/orgs/community/discussions/37267 - if people want to learn more about this line

**Review by adironene — APPROVED** [review]
Thank you for looking into this Kenneth. Changes look good. I can go ahead and change deployment option in github pages once this merges in

**Comment by kabirsachdev7** [comment]
Reviewed the pipeline and it looks good to be merged! Good work!

**Review by kabirsachdev7 — APPROVED** [review]
Reviewed the pipeline and it looks good to be merged! Good work!

**Pull request merged**